content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,520 @@
+ 
+I will be using the popular iris data set for the tutorial, you can get it almost anywhere but I downloaded this from [Kaggle](https://www.kaggle.com/vikrishnan/iris-dataset).
+ 
+Load the data into the notebook
+```python
+df = pd.read_csv("/content/drive/MyDrive/IRIS.csv")
+```
+```python
+df.head()
+```
+![df.head.png](/engineering-education/statistical-hypothesis-testing-python-implementation/dataset_head.PNG)
+On checking the first five, it seems all the species of flower are setosa, but the iris dataset has 2 more samples apart from that, to be sure, let's run df.sample() to visualize samples randomly.
+```python
+df.sample(10)
+```
+![sampleten](/engineering-education/statistical-hypothesis-testing-python-implementation/dataset_sample.PNG)
+ 
+Always check for missing values to avoid future complications.
+```python
+df.isnull().sum()
+```
+```bash
+sepal_length    0
+sepal_width     0
+petal_length    0
+petal_width     0
+species         0
+dtype: int64
+```
+We can be sure now that there are 3 classes of species and there are no missing values.
+Let's proceed with the data manipulation.
+ 
+Let's use petal width to compare species for our chi-square test.
+ 
+The first step is to check the summary of `petal_width`
+ 
+Secondly, use the information derived to convert it to a categorical variable.
+```python
+df.petal_width.describe()
+```
+```bash
+count    150.000000
+mean       1.198667
+std        0.763161
+min        0.100000
+25%        0.300000
+50%        1.300000
+75%        1.800000
+max        2.500000
+Name: petal_width, dtype: float64
+```
+Since, the operation we are going to be performing are mathematical operations and they don't work with letters or words except numbers.
+ 
+The Chi-square test helps in determining if there is a notable difference between the observed and normal frequencies in one or more categories, the values of petal_width must be changed into categories of zeros and ones using the 50% percentile which is the second quartile as the yardstick.
+ 
+There are clear [differences](https://byjus.com/maths/difference-between-percentage-and-percentile/) between percentage and percentile. As percentage is just a fraction of a hundred, the percentile is more concerned with ranks, it can be termed as the percentage of values that fall below a certain value mark.
+ 
+It is rather used to find the position of a value in a given set of values. For example, if a girl has the 50th percentile on an examination of 200, we can say that by scoring 200 marks, she has higher marks than 50% of the remaining class.
+There are two conditions that must be satisfied while calculating percentile.
+1. Data must be arranged in ascending order.
+2. if the index is not an integer, it must be rounded up.
+```
+np.percentile(df.petal_width, 50)
+```
```

Comment:
Do add the Python tag for this piece of code.
And please print the output. The appropriate code block would be - 

```python
print(np.percentile(df.petal_width, 50))
```